### PR TITLE
Dilpreet | Test | Combine mobile code and mobile number field in creating / update account api

### PR DESCRIPTION
**Pull request by dilpreetsinghofficial**
https://app.clickup.com/t/2khdvty

### CODE REVIEW

**Comment by dilpreetsinghofficial** [comment]
reviewed and please ignored custom CSS in scss format  due to not running on runtime

File: apps/web-giddh/src/app/shared/header/components/account-add-new-details/account-add-new-details.component.ts [hunk 1]
```diff
@@ -169,6 +169,19 @@ export class AccountAddNewDetailsComponent implements OnInit, OnChanges, AfterVi
...
     /** This will hold toggle buttons value and size */
     public bootstrapToggleSwitch = BootstrapToggleSwitch;
+    /** This will hold SearchCountryField */
+    public SearchCountryField = SearchCountryField;
+    /** This will hold CountryISO */
+    public CountryISO = CountryISO;
+    /** This will hold PhoneNumberFormat */
+    public PhoneNumberFormat = PhoneNumberFormat;
+    /** This will hold newCountryCode */
+    public newCountryCode = '';
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/shared/header/components/account-add-new-details/account-add-new-details.component.ts]: define type of value

File: apps/web-giddh/src/app/shared/header/components/account-add-new-details/account-add-new-details.component.ts [hunk 2]
```diff
@@ -169,6 +169,19 @@ export class AccountAddNewDetailsComponent implements OnInit, OnChanges, AfterVi
...
+    /** This will hold oldCountryCode */
+    public oldCountryCode = '';
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/shared/header/components/account-add-new-details/account-add-new-details.component.ts]: define type of value

File: apps/web-giddh/src/app/shared/header/components/account-add-new-details/account-add-new-details.component.ts [hunk 3]
```diff
@@ -349,7 +367,7 @@ export class AccountAddNewDetailsComponent implements OnInit, OnChanges, AfterVi
             openingBalanceType: ['CREDIT'],
             foreignOpeningBalance: [''],
             openingBalance: [''],
-            mobileNo: [''],
+            mobileNo: ['', Validators.required],
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/shared/header/components/account-add-new-details/account-add-new-details.component.ts]: shouldn't be required

File: apps/web-giddh/src/app/shared/header/components/account-update-new-details/account-update-new-details.component.ts [hunk 4]
```diff
@@ -214,6 +214,18 @@ export class AccountUpdateNewDetailsComponent implements OnInit, OnDestroy, OnCh
...
     /** This will hold toggle buttons value and size */
     public bootstrapToggleSwitch = BootstrapToggleSwitch;
+    /** This will hold SearchCountryField */
+    public SearchCountryField = SearchCountryField;
+    /** This will hold CountryISO */
+    public CountryISO = CountryISO;
+    /** This will hold PhoneNumberFormat */
+    public PhoneNumberFormat = PhoneNumberFormat;
+    /** This will hold newCountryCode */
+    public newCountryCode = '';
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/shared/header/components/account-update-new-details/account-update-new-details.component.ts]: define type of value

File: apps/web-giddh/src/app/shared/header/components/account-update-new-details/account-update-new-details.component.ts [hunk 5]
```diff
@@ -214,6 +214,18 @@ export class AccountUpdateNewDetailsComponent implements OnInit, OnDestroy, OnCh
...
+    /** This will hold oldCountryCode */
+    public oldCountryCode = '';
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/shared/header/components/account-update-new-details/account-update-new-details.component.ts]: define type of value

File: apps/web-giddh/src/app/shared/header/components/account-update-new-details/account-update-new-details.component.ts [hunk 6]
```diff
@@ -448,7 +465,7 @@ export class AccountUpdateNewDetailsComponent implements OnInit, OnDestroy, OnCh
             openingBalanceType: ['CREDIT'],
             foreignOpeningBalance: [''],
             openingBalance: [''],
-            mobileNo: [''],
+            mobileNo: ['', Validators.required],
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/shared/header/components/account-update-new-details/account-update-new-details.component.ts]: shouldn't be required

File: apps/web-giddh/src/app/shared/header/components/account-update-new-details/account-update-new-details.component.ts [hunk 7]
```diff
@@ -1767,4 +1782,20 @@ export class AccountUpdateNewDetailsComponent implements OnInit, OnDestroy, OnCh
         this.closeAccountModal.emit(true);
         this.store.dispatch(this.groupWithAccountsAction.HideAddAndManageFromOutside());
     }
+
+    /**
+ * This will check no and replace old country code with new country code
+ *
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/shared/header/components/account-update-new-details/account-update-new-details.component.ts]: format

File: apps/web-giddh/src/app/shared/header/components/account-add-new-details/account-add-new-details.component.ts [hunk 8]
```diff
@@ -169,6 +169,19 @@ export class AccountAddNewDetailsComponent implements OnInit, OnChanges, AfterVi
     public availableFieldTypes: any = FieldTypes;
     /** This will hold toggle buttons value and size */
     public bootstrapToggleSwitch = BootstrapToggleSwitch;
+    /** This will hold SearchCountryField */
+    public SearchCountryField = SearchCountryField;
```
Comment (the-prateek-mishra) [inline apps/web-giddh/src/app/shared/header/components/account-add-new-details/account-add-new-details.component.ts]:
```suggestion
    public searchCountryField = SearchCountryField;
```

File: apps/web-giddh/src/app/shared/header/components/account-add-new-details/account-add-new-details.component.ts [hunk 9]
```diff
@@ -169,6 +169,19 @@ export class AccountAddNewDetailsComponent implements OnInit, OnChanges, AfterVi
...
+    /** This will hold CountryISO */
+    public CountryISO = CountryISO;
```
Comment (the-prateek-mishra) [inline apps/web-giddh/src/app/shared/header/components/account-add-new-details/account-add-new-details.component.ts]:
```suggestion
    public countryISO = CountryISO;
```

File: apps/web-giddh/src/app/shared/header/components/account-add-new-details/account-add-new-details.component.ts [hunk 10]
```diff
@@ -169,6 +169,19 @@ export class AccountAddNewDetailsComponent implements OnInit, OnChanges, AfterVi
...
+    /** This will hold PhoneNumberFormat */
+    public PhoneNumberFormat = PhoneNumberFormat;
```
Comment (the-prateek-mishra) [inline apps/web-giddh/src/app/shared/header/components/account-add-new-details/account-add-new-details.component.ts]:
```suggestion
    public phoneNumberFormat = PhoneNumberFormat;
```

File: apps/web-giddh/src/app/shared/header/components/account-update-new-details/account-update-new-details.component.ts [hunk 11]
```diff
@@ -214,6 +214,18 @@ export class AccountUpdateNewDetailsComponent implements OnInit, OnDestroy, OnCh
     public availableFieldTypes: any = FieldTypes;
     /** This will hold toggle buttons value and size */
     public bootstrapToggleSwitch = BootstrapToggleSwitch;
+    /** This will hold SearchCountryField */
+    public SearchCountryField = SearchCountryField;
```
Comment (the-prateek-mishra) [inline apps/web-giddh/src/app/shared/header/components/account-update-new-details/account-update-new-details.component.ts]:
```suggestion
    public searchCountryField = SearchCountryField;
```

File: apps/web-giddh/src/app/shared/header/components/account-update-new-details/account-update-new-details.component.ts [hunk 12]
```diff
@@ -214,6 +214,18 @@ export class AccountUpdateNewDetailsComponent implements OnInit, OnDestroy, OnCh
...
+    /** This will hold CountryISO */
+    public CountryISO = CountryISO;
```
Comment (the-prateek-mishra) [inline apps/web-giddh/src/app/shared/header/components/account-update-new-details/account-update-new-details.component.ts]:
```suggestion
    public countryISO = CountryISO;
```

File: apps/web-giddh/src/app/shared/header/components/account-update-new-details/account-update-new-details.component.ts [hunk 13]
```diff
@@ -214,6 +214,18 @@ export class AccountUpdateNewDetailsComponent implements OnInit, OnDestroy, OnCh
...
+    /** This will hold PhoneNumberFormat */
+    public PhoneNumberFormat = PhoneNumberFormat;
```
Comment (the-prateek-mishra) [inline apps/web-giddh/src/app/shared/header/components/account-update-new-details/account-update-new-details.component.ts]:
```suggestion
    public phoneNumberFormat = PhoneNumberFormat;
```

File: apps/web-giddh/src/app/shared/header/components/account-update-new-details/account-update-new-details.component.ts [hunk 14]
```diff
@@ -1767,4 +1782,20 @@ export class AccountUpdateNewDetailsComponent implements OnInit, OnDestroy, OnCh
         this.closeAccountModal.emit(true);
         this.store.dispatch(this.groupWithAccountsAction.HideAddAndManageFromOutside());
     }
+
+    /**
```
Comment (the-prateek-mishra) [inline apps/web-giddh/src/app/shared/header/components/account-update-new-details/account-update-new-details.component.ts:1786]: Correct JS doc formatting

**Review by the-prateek-mishra — CHANGES_REQUESTED** [review]

**Review by ravinderwalkover — CHANGES_REQUESTED** [review]

**Review by the-prateek-mishra — APPROVED** [review]

**Review by ravinderwalkover — APPROVED** [review]

**Pull request merged**